### ISSUE TITLE
Underzooming tiles misalignment

### ISSUE THREAD

**Issue by neokore** [visgl/deck.gl#4965]
#### Description
When loading a tiled layer for the first time using a lower zoom than its `minZoom` it is rendered smaller and misaligned from the basemap, but after any prop refresh (e.g. zooming or panning the map) the layer will be adjusted and will fit the basemap. This bug mainly appears when using plain JS instead of React, for example.

#### Repro Steps
1. Set up a basemap and a `MVTLayer`.
2. Set the `minZoom`, `maxZoom` and `extent` props.
3. Set a zoom level lower than the set in `initialViewState`.
4. Load the map without zooming or panning and wait until all tiles render. The map should look like this:
![image](https://user-images.githubusercontent.com/938057/93870173-15a74380-fccd-11ea-928e-3a9263028237.png)
5. Drag the map. Now the map should look like this:
![image](https://user-images.githubusercontent.com/938057/93870242-2fe12180-fccd-11ea-9e2c-c6172e75bddd.png)

You can also reproduce the issue with this gist: https://gist.github.com/neokore/611492fa00ef446df97ad4eb4b76163b
And check that using some JS frameworks, e.g. React or whatever Codepen uses, it works OK: https://codepen.io/neokore/pen/KKzrOPd

#### Environment (please complete the following information):
 - Framework Version: deck.gl 8.2.8
 - Browser Version: Chrome 85.0.4183.121
 - OS: Linux

#### Logs
There is no message on console.

**Comment by neokore** [comment]
I'm investigating it from my side and I'll try to send you a PR as soon as I find a fix.

**Comment by Pessimistress** [comment]
This issue can be reproduced on CodePen if you set the initial `latitude` to `40`.

This is because Mapbox does not allow white space above or below the map. The map is automatically shifted so that the whole viewport is covered. The `MapController` performs the same normalization when you interact with the map:

https://github.com/visgl/deck.gl/blob/7dd6a75cd5b5036f0f39f94b610f017a16c14f32/modules/core/src/controllers/map-controller.js#L374

But it never checked the initial state.

A possible fix is to run normalization whenever viewport dimension changes. react-map-gl does so in its controller:

https://github.com/visgl/react-map-gl/blob/83933663dc238d4052f0be7011c75ae0e85daee5/src/utils/map-controller.js#L180-L187

**Comment by Pessimistress** [comment]
Before this is addressed, a workaround in your application is to always set the initial `latitude` to `0` at low zoom levels. Bigger numbers will be overwritten anyways.

**Comment by neokore** [comment]
Thank you, I thought that it was related to MVTLayer.
And could make sense to do the same as in react-map-gl by default in the startup of the map for Deck.GL? Taking into account that everyday display resolutions are going higher and higher, it would be great to avoid devs to add this line for every map they create in plain JS.

**Comment by Pessimistress** [comment]
The behavior should not change to display resolutions. It's dependent on the aspect ratio of your viewport.

Regardless, I agree we should fix this in the controller. Feel free to open a PR if you need this soon.

**Comment by zakjan** [comment]
This is an opinonated choice of Mapbox map library. However, there are other map libraries which allow zooming out: officially supported ArcGIS, Google Maps, also unofficially supported Leaflet with my recently published deck.gl-leaflet.

I would suggest to remove the normalization from MapController, normalize in react-map-gl only.

**Comment by Pessimistress** [comment]
`@deck.gl/arcgis` and `@deck.gl/google-maps` do not use deck.gl's built-in controller. If you are rendering deck as a leaflet layer you shouldn't be using MapController either.

**Comment by zakjan** [comment]
Right, I don't use it as well. If removing this behavior from deck.gl's built-in controller is complicated, could it be configurable instead?

**Comment by Pessimistress** [comment]
You are correct that this is an opinionated choice, though this is the choice of deck.gl, which works independently from Mapbox. When you use an external controller, deck does not normalize the view state.

Removing the behavior will be a major breaking change. We are open to adding an option to turn it off.

**Comment by zakjan** [comment]
Thanks for confirming, I'll make a PR.

**Comment by Pessimistress** [comment]
Fix is available in `8.5.0-alpha.5`
